tests/rust/chorded/overlapping.rs

```diff
@@ -0,0 +1,164 @@
+use smart_keymap::input;
+use smart_keymap::key;
+use smart_keymap::keymap;
+use smart_keymap::tuples;
+
+use keymap::DistinctReports;
+use keymap::Keymap;
+
+use key::{chorded, composite, keyboard};
+use tuples::Keys4;
+
+type Ctx = composite::Context;
+type Ev = composite::Event;
+type PKS = composite::PendingKeyState;
+type KS = composite::KeyState;
+type CK = composite::ChordedKey<composite::Layered<composite::TapHold<keyboard::Key>>>;
+type AK = composite::ChordedKey<composite::Layered<composite::TapHold<keyboard::Key>>>;
+
+// 4-key keymap
+//   A B C D
+// chords:
+// 0: X X - - => M=0x10
+// 1: X X X - => N=0x11
+// 2: - X X - => O=0x12
+// 3: - - X X => P=0x13
+
+const KEYS: Keys4<CK, CK, CK, AK, Ctx, Ev, PKS, KS> = tuples::Keys4::new((
+    composite::ChordedKey::Chorded(chorded::Key::new(
+        &[
+            (
+                0,
+                composite::Layered(composite::TapHold(keyboard::Key::new(0x10))),
+            ),
+            (
+                1,
+                composite::Layered(composite::TapHold(keyboard::Key::new(0x11))),
+            ),
+        ],
+        composite::Layered(composite::TapHold(keyboard::Key::new(0x04))),
+    )),
+    composite::ChordedKey::Chorded(chorded::Key::new(
+        &[(
+            2,
+            composite::Layered(composite::TapHold(keyboard::Key::new(0x12))),
+        )],
+        composite::Layered(composite::TapHold(keyboard::Key::new(0x05))),
+    )),
+    composite::ChordedKey::Chorded(chorded::Key::new(
+        &[(
+            3,
+            composite::Layered(composite::TapHold(keyboard::Key::new(0x13))),
+        )],
+        composite::Layered(composite::TapHold(keyboard::Key::new(0x06))),
+    )),
+    composite::ChordedKey::Auxiliary(chorded::AuxiliaryKey::new(composite::Layered(
+        composite::TapHold(keyboard::Key::new(0x07)),
+    ))),
+));
+
+const CONTEXT: Ctx = key::composite::Context::from_config(composite::Config {
+    chorded: chorded::Config {
+        chords: [
+            Some(chorded::ChordIndices::from_slice(&[0, 1])),
+            Some(chorded::ChordIndices::from_slice(&[0, 1, 2])),
+            Some(chorded::ChordIndices::from_slice(&[1, 2])),
+            Some(chorded::ChordIndices::from_slice(&[2, 3])),
+        ],
```

Comment:
n.b. In order to test against more than 4 chords, it'd be convenient to use the `Slice` for the `key::chorded::Config`, too.